File: Segundo Curso/1er cuatrimestre/PDOO/Teoria/Pruebas/AplicacionPerros/src/main/java/aplicacionperro/Perro.java

package aplicacionperros;

public class Perro {
    private String raza;
    private int peso;
    boolean agresivo; /*Si no pongo nada delante significa que es de paquete,
                       que es visible para otros paquetes*/
    public String nombre;
    
    public Perro(String raza, int p, String n){ // Constructor (dar valor a todos los atributos
        this.raza = raza;                     // Si se le pone this, se refiere al atributo
        peso = p;
        agresivo = false;
        nombre = n;
    }
    
    public void getNombre(){ //Metodo consultor (get delante)
        return nombre;
    }
    
    public void ladrar(){ // Comportamiento para objeto de la clase
        System.out.println("guau guau");
    }
    
    public void setNombre(String nuevoNombre){ // Modificador
        nombre = nuevoNombre;
    }
    
    public void ponteAgresivo() {  
        agresivo = true;
    }
    
    public boolean getAgresivo(){
        return agresivo;
    }
    
    
}

File: Segundo Curso/1er cuatrimestre/PDOO/Teoria/Pruebas/AplicacionPerrosRuby/lib/perro.rb

class Perro
  
  attr_accessor :nombre, :agresivo   #consultor y modificador
  
  attr_reader :peso  #consultor solo
  
  attr_writer :raza  #modificador solo
  
  def initialize(raza, peso, agresivo, nombre)  #todos los atributos son privados y no se pueden cambiar
    @raza=raza
    @peso=peso
    @agresivo=agresivo
    @nombre=nombre  
  end
  

  def ladrar
    "guau guau"
  end
  
  
  def get_peso  #consultor de peso
    @peso   #importante no olvidar @ para atributos. No da error, piensa que son variables locales
  end
  
  
  private :get_peso, :nombre=
  
end
